polaris-react/src/components/LegacyCard/LegacyCard.scss

```diff
@@ -58,7 +58,7 @@
     padding: var(--p-space-5) var(--p-space-5) 0;
 
     #{$se23} & {
-      padding: var(--p-space-4) var(--p-space-4) 0;
+      padding: var(--p-space-2) var(--p-space-4);
```

Comment:
Treat `.Header` like any other section with `8px` top and bottom, `16px` sides. We now apply the extra top padding in `js` in the `tsx` file if the conditions are correct (i.e. if it is a first child of `LegacyCard`)